flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -426,58 +807,120 @@ def add_spw_forcing(
+            simulation_paths.append(
+                results_path.joinpath(
+                    "simulations",
+                    self._site.attrs.sfincs.overland_model,
+                )
+            )
+            # Create a folder name for the offshore model (will not be used if offshore model is not created)
+            simulation_paths_offshore.append(
+                results_path.joinpath(
+                    "simulations",
+                    self._site.attrs.sfincs.offshore_model,
+                )
+            )
 
-    def turn_off_bnd_press_correction(self):
-        self.sf_model.set_config("pavbnd", -9999)
+        # TODO investigate
+        elif mode == Mode.risk:  # risk mode requires an additional folder layer
+            for subevent in event.get_subevents():
+                simulation_paths.append(
+                    results_path.joinpath(
+                        "simulations",
+                        subevent.attrs.name,
+                        self._site.attrs.sfincs.overland_model,
+                    )
+                )
 
-    def add_obs_points(self):
-        """Add observation points provided in the site toml to SFINCS model."""
-        if self.site.attrs.obs_point is not None:
-            obs_points = self.site.attrs.obs_point
-            names = []
-            lat = []
-            lon = []
-            for pt in obs_points:
-                names.append(pt.name)
-                lat.append(pt.lat)
-                lon.append(pt.lon)
+                # Create a folder name for the offshore model (will not be used if offshore model is not created)
+                simulation_paths_offshore.append(
+                    results_path.joinpath(
+                        "simulations",
+                        subevent.attrs.name,
+                        self._site.attrs.sfincs.offshore_model,
+                    )
+                )
 
-            # create GeoDataFrame from obs_points in site file
-            df = pd.DataFrame({"name": names})
-            gdf = gpd.GeoDataFrame(
-                df, geometry=gpd.points_from_xy(lon, lat), crs="EPSG:4326"
-            )
+        return simulation_paths, simulation_paths_offshore
 
-            # Add locations to SFINCS file
-            self.sf_model.setup_observation_points(locations=gdf, merge=False)
+    def _get_flood_map_path(self) -> list[Path]:
+        """_summary_."""
+        results_path = self._get_result_path()
+        mode = Database().events.get(self._scenario.attrs.event).get_mode()
+
+        if mode == Mode.single_event:
+            map_fn = [results_path.joinpath("max_water_level_map.nc")]
 
-    def read_zsmax(self):
+        elif mode == Mode.risk:
+            map_fn = []
+            for rp in self._site.attrs.risk.return_periods:
+                map_fn.append(results_path.joinpath(f"RP_{rp:04d}_maps.nc"))
+
+        return map_fn
+
+    def _get_wl_df_from_offshore_his_results(self) -> pd.DataFrame:
```

Comment:
should we do this in one function or create a HazardOutput object for the offshore model first and treat this as a Timeseries input object to the overland model?